### ISSUE TITLE
dumb rotate isn't affecting jump nodes

### ISSUE THREAD

**Issue by AxemP** [scp-fs2open/fs2open.github.com#1682]
We used to be able to pull off some neat things with dumb rotate, like rotating jump nodes, or subtly moving clouds in a skybox. This appears to not work anymore. By my very rough code reading skills in modelread.cpp, it seems it's only allowed for ships now? Can this be widened to all models once again?

Please see sample mod. The everything (except the player) has a dumb rotate part to it. So part of the skybox should move, the jumpnode should move. The candy bar already rotates, but that's there to show that dumb rotate isn't completely broken. Just kinda.

[dumbrotate.zip](https://github.com/scp-fs2open/fs2open.github.com/files/1905402/dumbrotate.zip)

**Comment by Goober5000** [comment]
This could have been caused by my changes in #512, but that PR received extensive review and non-ship rotating models were specifically mentioned.

**Comment by Goober5000** [comment]
From the #scp chat, it looks like @TRBlount is looking at it.  My assumption is that any model which has a $dumb_rotate (or $dumb_rotate_time) property will have the intrinsic-rotate flag automatically assigned, but I won't have time to look at this until tomorrow.

**Comment by Goober5000** [comment]
My assumption turned out to be correct.  Here's the relevant code excerpt from modelread.cpp...

```
				// note, this should come BEFORE do_new_subsystem() for proper error handling (to avoid both rotating and dumb-rotating submodel)
				int idx = prop_string(props, &p, "$dumb_rotate_time", "$dumb_rotate_rate", "$dumb_rotate");
				if (idx >= 0) {
					pm->submodel[n].movement_type = MOVEMENT_TYPE_INTRINSIC_ROTATE;
					pm->flags |= PM_FLAG_HAS_INTRINSIC_ROTATE;
```

So the bug must be caused by something else, e.g. the model instance creation that someone mentioned on Discord.

Do we know when this bug was first introduced?

**Comment by Goober5000** [comment]
@AxemP can you see if you can narrow down the timeframe when this bug first appeared?

**Comment by FSCyborg** [comment]
So, I just took a good look at everything, and the reason why jump nodes do not use intrinsic rotate is because jump nodes are not added to the polymodel instance list.  I just finished a prototype of the a patch adding that functionality, and I need to test it this weekend.  (It required a few functions be added to the jumpnode class) 

I am still trying to figure out why skyboxes are not moving.  I don't like how roundabout the check is, but I can't yet see how it could be broken.

**Comment by Goober5000** [comment]
Excellent progress!

**Comment by AxemP** [comment]
I've forgotten about this issue until its just come up again. Yay for finding how jump nodes broke, it saves me from finding out exactly *when* they broke.

For skyboxes, they are working... I thought maybe somehow they got fixed, but I just looked at that test mod I linked to and... the rotating part had a rotation axis of **none**. So... d'oh.

**Comment by FSCyborg** [comment]
Oh good!  That means I'm not crazy! I'll try to build the jump nide fix tomorrow and test.

**Comment by FSCyborg** [comment]
Successfully tested!  #2136 is the pull request.  And let's see if it'll merge easily.

**Comment by Goober5000** [comment]
Fixed via #2136
